File: 프로그래머스/연습문제/Lv2/연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>
using namespace std;

int solution(vector<int> elements) {
    // 수열 두 배로 붙여넣기
    vector<int> twice_ele = elements;
    for(int e : elements) twice_ele.push_back(e);
    
    set<int> nums;
    for(int i=1; i<=elements.size(); i++){
        for(int j=0; j<elements.size(); j++){
            int sum = 0;
            for(int k=j; k<j+i; sum+=twice_ele[k++]);
            nums.insert(sum);
        }
    }
    
    return nums.size();
}

File: 프로그래머스/연습문제/Lv2/연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    twice_ele = []
    for e in elements: twice_ele.append(e)
    for e in elements: twice_ele.append(e)

    nums = set()
    for i in range(1, len(elements)+1):
        for j in range(len(elements)):
            nums.add(sum(twice_ele[j:j+i]))

    return len(nums)
